src/deterministic_secure_broadcast.rs

```diff
@@ -341,10 +324,22 @@ impl<A: SecureBroadcastAlgorithm> SecureBroadcastProc<A> {
     }
 
     fn validate_payload(&self, from: Actor, payload: &Payload<A::Op>) -> bool {
+
+        match payload {
+            Payload::RequestValidation { msg } if msg.dot != self.received.inc(from) => {
+                println!("[DSB] validating RequestValidation: not the next message. msg.dot = {:?}  expecting: {:?}", msg.dot, self.received.inc(from));
+            },
+            Payload::ProofOfAgreement { msg, .. } if msg.dot != self.delivered.inc(from) => {
+                println!("[DSB] validating ProofOfAgreement: not the next message. msg.dot = {:?}  expecting: {:?}", msg.dot, self.delivered.inc(from));
+            },
+            _ => {}
+        }
+
         let validation_tests = match payload {
             Payload::RequestValidation { msg } => vec![
                 (from == msg.dot.actor, "source does not match the msg dot"),
-                (msg.dot == self.received.inc(from), "not the next msg"),
+// Experimental: removing this check temporarily.  probably not OK to remove.
+//                (msg.dot == self.received.inc(from), "not the next msg"),
```

Comment:
This would break our double spend protection

Reply:
yeah, I knew you wouldn't like that one.  :-)